### PR TITLE
Add dark theme using QDarkStyleSheet

### PR DESCRIPTION
**Pull request by tmontaigu**
This adds to CC a Dark (and also a light) theme
using the QDarkStyleSheet (https://github.com/ColinDuquesnoy/QDarkStyleSheet)

This dark theme is not perfect but a bit better than not having not dark theme at all for platforms that do not have one by default like Windows.

### CODE REVIEW

**Comment by dgirardeau** [comment]
Would it be possible to integrate it as a submodule instead?

**Comment by tmontaigu** [comment]
Yes

File: libs/CCAppCommon/src/ccApplicationBase.cpp [hunk 1]
```diff
@@ -273,15 +273,41 @@ void ccApplicationBase::setupPaths()
 
 bool ccApplicationBase::setAppStyle(QString styleKey)
 {
-	QStyle* style = QStyleFactory::create(styleKey);
-	if (!style)
-	{
-		ccLog::Warning("Invalid style key or style couldn't be created: " + styleKey);
-		return false;
-	}
+	const auto loadStyleSheet = [this](const QString& ressourcePath) {
```
Comment (dgirardeau) [inline libs/CCAppCommon/src/ccApplicationBase.cpp:276]: It's to force me to apply the clang format to the whole project? 😆 

For now can you manually move all these '{' to the next line?

File: .gitmodules [hunk 2]
```diff
@@ -30,3 +30,6 @@
 	path = plugins/core/Standard/q3DMASC
 	url = https://github.com/dgirardeau/q3DMASC.git
 	branch = master
+[submodule "libs/CCAppCommon/QDarkStyleSheet"]
```
Comment (dgirardeau) [inline .gitmodules:33]: Thanks!

File: libs/CCAppCommon/src/ccApplicationBase.cpp [hunk 3]
```diff
@@ -273,15 +273,41 @@ void ccApplicationBase::setupPaths()
...
+		QStyle* style = QStyleFactory::create(styleKey);
+		if (!style)
+		{
+			ccLog::Warning("Invalid style key or style couldn't be created: " + styleKey);
+			return false;
+		}
 
-	ccLog::Print("Applying application style: " + styleKey);
-	setStyle(style);
+		setStyleSheet("");
```
Comment (dgirardeau) [inline libs/CCAppCommon/src/ccApplicationBase.cpp:307]:
```suggestion
		setStyleSheet(QString());
```

or maybe setStyleSheet({}); if it works

File: libs/CCAppCommon/src/ccDisplayOptionsDlg.cpp [hunk 4]
```diff
@@ -91,26 +93,33 @@ ccDisplayOptionsDlg::ccDisplayOptionsDlg(QWidget* parent)
...
+		// Which is necessarily the key currently stored in persistent settings
+		// since this is saved each time the current setting is selected,
+		// and since it's the first time the user opens the settings for this session,
+		// then, the style was not changed.
+		QSettings settings;
+		settings.beginGroup(ccPS::AppStyle());
+		const QString defaultStyleName = settings.value("style").toString();
+		settings.endGroup();
+
+				// fill the combo-box
```
Comment (dgirardeau) [inline libs/CCAppCommon/src/ccDisplayOptionsDlg.cpp:106]: strange formatting or is it Github visualization?

File: libs/CCAppCommon/src/ccApplicationBase.cpp (hunk 1 above)
Comment (tmontaigu) [reply libs/CCAppCommon/src/ccApplicationBase.cpp:276]: :wink:

File: libs/CCAppCommon/src/ccDisplayOptionsDlg.cpp (hunk 4 above)
Comment (tmontaigu) [reply libs/CCAppCommon/src/ccDisplayOptionsDlg.cpp:106]: strange formatting

File: libs/CCAppCommon/src/ccApplicationBase.cpp [hunk 5]
```diff
@@ -273,15 +273,51 @@ void ccApplicationBase::setupPaths()
...
-		ccLog::Warning("Invalid style key or style couldn't be created: " + styleKey);
-		return false;
+		QFile f(resourcePath);
+		if (!f.exists())
+		{
+			f.close();
+			return false;
+		}
+		else
+			{
```
Comment (dgirardeau) [inline libs/CCAppCommon/src/ccApplicationBase.cpp:285]: shifted

File: libs/CCAppCommon/src/ccDisplayOptionsDlg.cpp [hunk 6]
```diff
@@ -91,26 +93,33 @@ ccDisplayOptionsDlg::ccDisplayOptionsDlg(QWidget* parent)
...
-		QStringList appStyles = QStyleFactory::keys();
-		for (int i = 0; i < appStyles.size(); ++i)
-		{
-			const QString& style = appStyles[i];
-			m_ui->appStyleComboBox->addItem(style);
+		m_ui->appStyleComboBox->addItem(QStringLiteral("QDarkStyleSheet::Light"));
+		m_ui->appStyleComboBox->addItem(QStringLiteral("QDarkStyleSheet::Dark"));
 
-			if (style.compare(defaultAppStyle, Qt::CaseInsensitive) == 0)
+		for (int i = 0; i < m_ui->appStyleComboBox->count(); ++i) {
```
Comment (dgirardeau) [inline libs/CCAppCommon/src/ccDisplayOptionsDlg.cpp:116]: one miss 😉

**Comment by dgirardeau** [comment]
Almost there 😄

**Review by dgirardeau — DISMISSED** [review]

**Comment by tmontaigu** [comment]
There was a change in the signature of ccApplicationBase::setAppstyle that should not have been in the coomit

**Review by dgirardeau — APPROVED** [review]

**Pull request merged**